src/components/modals/FormStepSaveModal.js

```diff
@@ -0,0 +1,187 @@
+import HelpText from 'components/HelpText';
+import Label from 'components/Label';
+import Modal from 'components/modals/Modal';
+import Input from 'components/Input';
+import {Toolbar, ToolbarList} from 'components/Toolbar';
+import Button from 'components/Button';
+import {getBEMClassName} from 'utils';
+import {useImmerReducer} from "use-immer";
+import Loader from "../Loader";
+
+
+const initialState = {
+  email: "",
+  errorMessage: "",
+  isSaving: false,
+};
+
+const reducer = (draft, action) => {
+  switch(action.type) {
+    case 'SET_EMAIL': {
+      draft.email = action.payload;
+      break;
+    }
+    case 'SET_ERROR_MESSAGE': {
+      draft.errorMessage = action.payload;
+      break;
+    }
+    case 'SET_IS_SAVING_TRUE': {
+      draft.isSaving = true;
+      break;
+    }
+    case 'SET_IS_SAVING_FALSE': {
+      draft.isSaving = false;
+      break;
+    }
+    default: {
+      throw new Error(`Unknown action ${action.type}`);
+    }
+  }
+};
+
+
+const FormStepSaveModal = ({
+    isOpen,
+    closeModal,
+    stepData,
+    saveStepDataUrl,
+    suspendFormUrl,
+}) => {
+  const history = useHistory();
+  const intl = useIntl();
+  const config = useContext(ConfigContext);
+
+  // const [email, setEmail] = useState("");
+  // const [errorMessage, setErrorMessage] = useState("");
+
+  const [
+    {email, errorMessage, isSaving},
+    dispatch
+  ] = useImmerReducer(reducer, initialState);
+
+  const onSubmit = async (event) => {
+    event.preventDefault();
+    dispatch({type: 'SET_IS_SAVING_TRUE'});
+    dispatch({type: 'SET_ERROR_MESSAGE', payload: ''});
+    let response = await put(saveStepDataUrl, {data: stepData});
+    if (!response.ok) {
+      dispatch({
+        type: 'SET_ERROR_MESSAGE',
+        payload: intl.formatMessage({
+          description: "Modal saving data failed message",
+          defaultMessage: "Saving the data failed, please try again later"
+        })
+      });
+      dispatch({type: 'SET_IS_SAVING_FALSE'});
+      return;
+    }
+    response = await post(suspendFormUrl, {email});
+    if (!response.ok){
+      dispatch({
+        type: 'SET_ERROR_MESSAGE',
+        payload: intl.formatMessage({
+          description: "Modal suspending form failed message",
+          defaultMessage: "Suspending the form failed, please try again later"
+        })
+      });
+      dispatch({type: 'SET_IS_SAVING_FALSE'});
+      return;
+    }
+    try {
+      // Destroy throws an exception if the API is not successful
+      await destroy(`${config.baseUrl}authentication/session`);
+    } catch (e) {
+      dispatch({
+        type: 'SET_ERROR_MESSAGE',
+        payload: intl.formatMessage({
+          description: "Modal logging out failed message",
+          defaultMessage: "Logging out failed, please try again later"
+        })
+      });
+      dispatch({type: 'SET_IS_SAVING_FALSE'});
+      return;
+    }
+    dispatch({type: 'SET_IS_SAVING_FALSE'});
```

Comment:
```suggestion
    dispatch({type: 'SAVE_SUCCEEDED'});
```
& handle all the relevant side effects in there (= reset the state back to initial state, basically)